### ISSUE TITLE
NullReferenceException when argument implements IEnumerable

### ISSUE THREAD

**Issue by FrolYo** [moq/moq4#169]
if I make IInput not implementing IEnumerable, then NullRef does not occur.

```
using System.Collections.Generic;

using Microsoft.VisualStudio.TestTools.UnitTesting;

using Moq;

[TestClass]
public class FooTest
{
    [TestMethod]
    public void ShouldCalculate()
    {
        var mocks = new MockRepository(MockBehavior.Loose);
        var service = mocks.Create<IDependency>();

        var input1 = mocks.OneOf<IInput>();
        service.Setup(x => x.Run(input1)).Returns(1);

        var input2 = mocks.OneOf<IInput>();
        service.Setup(x => x.Run(input2)).Returns(2);

        var foo = new Foo(service.Object);
        foo.Calculate(input1);
        foo.Calculate(input2);
    }
}

public interface IDependency
{
    decimal Run(IInput input);
}

public interface IInput : IEnumerable<int>
{
}

public class Foo
{
    private readonly IDependency dependency;

    public Foo(IDependency dependency)
    {
        this.dependency = dependency;
    }

    public void Calculate(IInput input)
    {
        this.dependency.Run(input);
    }
}
```

**Comment by stakx** [comment]
This appears to be remotely related to #22. Except here, the mock apparently has a setup for `IEnumerable.GetEnumerator`, but not for `IEnumerable<T>.GetEnumerator` (and not the other way around).

**Comment by stakx** [comment]
Note to self: This error is due to `Interceptor.AddCall` creating an `ExpressionKey`, which involves converting the method setup to a string representation using `ExpressionStringBuilder`, which in turn tries to enumerate the mock object (and since `IEnumerable` isn't correctly set up, this will fail due to a small bug). Getting rid of the string representation would help (as suggested by https://github.com/moq/moq4/issues/328#issuecomment-285655952) but I'm not sure how that can be done. The simplest solution is to fix the enumeration in `ExpressionStringBuilder`.

**Comment by stakx** [comment]
This will be fixed in the next release of Moq (version >4.7.63).
